scripts/apps/dashboard/world-clock/timezones-all-labels.ts

```diff
@@ -1,6 +1,16 @@
 import {gettext} from 'core/utils';
 
-export const getTimezoneLabels = () => ({
+export const getTimezoneLabel = (tz) => {
+    const labels = getTimezoneLabels();
+
+    if (labels.hasOwnProperty(tz)) {
+        return labels[tz];
+    }
+
+    return tz;
```

Comment:
It would be good to log a warning to sentry if it isn't found. Why wouldn't it be found? Is the list incomplete or what?

Reply:
This way is more useful for example to make tests with non-existing timezones and not having to mockup all the labels

Reply:
Why not import real labels and use them in tests?

Reply:
I meant when you create dummy names like [here](https://github.com/superdesk/superdesk-client-core/blob/d881ecd0bd1186c5ee6d6f34ae965ef4fbe4698e/scripts/apps/dashboard/world-clock/tests/tzdata.spec.ts#L66)

Reply:
Can't real values be used instead of those dummy names?

Reply:
But is there any issues with the current implementation?

Reply:
Code meant only for testing **inside implementation** is the issue.

We can restructure code by splitting blocks into functions so it's easier to test, but we shouldn't change the logic for the sake of testing.

Your version doesn't look as bad as the snippet below, but it is the same in principle.

```
export const getTimezoneLabel = (tz) => {
    if (isBeingTested) {
        return 'test label';
    }

    return getTimezoneLabels()[tz];
}
```

In your case, you are adding a conditional branch too, which is only meant for testing, only that it's not obvious that it's only for testing.

The best solution is simply passing valid data, so no exceptions are required as to how it would run in production.

If that's not possible or hard, you can mock the entire function:

```
// implementation code

const getLabel = (tz) => getTimezoneLabels()[tz];

export const timezone = {
    getLabel: getLabel,
};


// test code

timezone.getLabel = () => 'test label';

// ...
```

Reply:
But why is my first implementation `getTimezoneLabels()[tz]` better than the current one `getTimezoneLabel(tz)`?

Also this one is more flexible because using a function could allow passing arguments like `(noWhitespace = true)` or something like that (not that we need it, but I don't see any drawback yet). The fact that is easier to write tests for it is just a side effect, not the only purpose of it.

Reply:
I'm not suggesting to not use a function - only that the function shouldn't contain test-specific logic. In my example bellow `// implementation code` comment, I'm using a function too.

> The fact that is easier to write tests for it is just a side effect, not the only purpose of it.

When `labels.hasOwnProperty(tz)` evaluates to `false` except for tests? If the answer is never, then it means that piece of code is there only for testing.

Reply:
Right now it only triggers for tests but I can totally imagine a feature where users create their own Timezones to display

Reply:
Code should represent how an application works currently, not potentially. If we regularly add logic "just in case we need it someday", we'd not be able to differentiate between production code and other pieces that never run and are there only "for future use".

It'd make it harder to understand how the application works and when refactoring we would have to spend time maintaining that future code.

This is not a big issue, but a code smell that doesn't seem hard to fix. Can you do it?

Reply:
fixed

Reply:
Great, thanks.